classes/Infrastructure/Auth/SentinelIdentityProvider.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace OpenCFP\Infrastructure\Auth;
+
+use Cartalyst\Sentinel\Sentinel;
+use OpenCFP\Domain\Model\User;
+use OpenCFP\Domain\Services\IdentityProvider;
+use OpenCFP\Domain\Services\NotAuthenticatedException;
+use OpenCFP\Domain\Speaker\SpeakerRepository;
+
+final class SentinelIdentityProvider implements IdentityProvider
+{
+    private $sentinel;
+    private $repo;
+
+    public function __construct(Sentinel $sentinel, SpeakerRepository $repo)
+    {
+        $this->sentinel = $sentinel;
+        $this->repo     = $repo;
+    }
+
+    /**
+     * Retrieves the currently authenticate user's username.
+     *
+     * @throws NotAuthenticatedException
+     *
+     * @return User
+     */
+    public function getCurrentUser()
+    {
+        $user = $this->sentinel->getUser();
+        if ($user == null || !$user) {
```

Comment:
Just `if (!$user)` is equivalent.

Reply:
Thanks, i updated it to the following, making it a bit more accurate
```php
if (!$user instanceof \Cartalyst\Sentinel\Users\UserInterface) {
    throw new NotAuthenticatedException();
}
```